app/upgrades/v17/upgrades.go

```diff
@@ -128,3 +142,40 @@ func CreateUpgradeHandler(
 		return migrations, nil
 	}
 }
+
+// FlipTwapSpotPriceRecords flips the denoms and spot price of twap record of a given pool.
+func FlipTwapSpotPriceRecords(ctx sdk.Context, pools []poolmanagertypes.PoolI, keepers *keepers.AppKeepers) error {
+	for _, pool := range pools {
+		twapRecordHistoricalPoolIndexed, err := keepers.TwapKeeper.GetAllHistoricalPoolIndexedTWAPsForPoolId(ctx, pool.GetId())
```

Comment:
nit: for future ref I believe its slightly better to assign poolId to a var if its used more than once in the method.

Reply:
yup will do that here